src/Zicht/Bundle/SolrBundle/Solr/QueryBuilder/Update.php

```diff
@@ -38,7 +38,7 @@ public function add($document, $params = [])
             'add',
             ['doc' => array_map(
                 function ($v) {
-                    if ($v instanceof \DateTime) {
+                    if ($v instanceof \DateTimeInterface) {
```

Comment:
This will not work as the `DateHelper::formatDate` will call `setTimezone` to change the time zone to UTC so it can be formatted to ISO-8601.

Second problem is that `DateTimeInterface` does not declare the `setTimezone` and `DateTimeImmutable` will ignore the call to this method which make DateHelper give unsuspected results.

So the DateHelper should be updated to support this but I would recommend to use `DateTime` or let your object return an ISO-8601 string (to be more immutable). This problem is also  fixed in the newer version (5.x)